ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpTreeBuilderBase.fs

```diff
@@ -117,32 +117,24 @@ type FSharpTreeBuilderBase(lexer, document: IDocument, lifetime, path: VirtualFi
     member x.AdvanceToTokenOrRangeEnd(tokenType: TokenNodeType, range: range) =
         x.AdvanceToTokenOrPos(tokenType, range.End)
 
-    member x.AdvanceToXmlDoc(xmlDoc: XmlDoc, maxPos: pos) =
-        x.AdvanceToStart(xmlDoc.Range)
-
-        // todo: add separate token type for xmlDoc comment?
-        let maxOffset = x.GetOffset(maxPos)
-        while x.CurrentOffset < maxOffset && not (x.TokenType == FSharpTokenType.LINE_COMMENT && startsWith "///" (x.Builder.GetTokenText())) do
-            x.AdvanceLexer()
+    member x.AdvanceToXmlDocOwner(xmlDoc: XmlDoc, expectedType: TokenNodeType, declarationRange: range) =
+        let range = xmlDocOwnerStartRange xmlDoc declarationRange
+        x.AdvanceToTokenOrRangeStart(expectedType, range)
 
     member x.AdvanceToTokenOrPos(tokenType: TokenNodeType, pos: pos) =
         let maxOffset = x.GetOffset(pos)
         while x.CurrentOffset < maxOffset && x.TokenType != tokenType && not x.Eof do
             x.AdvanceLexer()
 
-    // todo: wrong ranges (e.g. missing union case bar)
-    // todo wrong order:
-    // todo multiple blocks:
```

Comment:
The multiple blocks todo is removed but I don't see any test case covering it. Was it removed by mistake? I'd expect this todo to be fixed in this PR.

Reply:
We already have some tests that cover this case:

[One](https://github.com/JetBrains/resharper-fsharp/blob/9321ca1ccbc4d356ebb2737ce79d800c0bd9e688/ReSharper.FSharp/test/data/parsing/Module%20-%20Top%20level%20-%20XmlDoc%2003%20-%20Multiple%20blocks.fs.gold), [two](https://github.com/JetBrains/resharper-fsharp/blob/9321ca1ccbc4d356ebb2737ce79d800c0bd9e688/ReSharper.FSharp/test/data/parsing/Type%20decl%20-%20XmlDoc%2004%20-%20Multiple%20blocks.fs.gold), [three](https://github.com/JetBrains/resharper-fsharp/blob/9321ca1ccbc4d356ebb2737ce79d800c0bd9e688/ReSharper.FSharp/test/data/parsing/Module%20-%20Nested%20-%20XmlDoc%2002.fs.gold)

Reply:
Thanks. Would be great to have something like this too:
```
/// Xml 1
// Comment
/// Xml 2
```

The other existing cases aren't parsed as xml comments for the declarations anymore. Since you've implemented a better logic in the parser, can we rename the tests, so they aren't about multiple blocks anymore?